src/libsyntax/tokenstream.rs

```diff
@@ -123,7 +123,11 @@ impl TokenTree {
     }
 
     pub fn joint(self) -> TokenStream {
-        TokenStream::JointTree(self)
+        TokenStream::Tree(self, Joint)
+    }
+
+    pub fn stream(self, is_joint: IsJoint) -> TokenStream {
```

Comment:
This means could probably be `into_stream`.

Reply:
Looking more at all the uses, maybe spelling out `TokenStream::Tree` explicitly wouldn't be a bad idea?

Especially for a `match` arm with `Tree` somewhere in the pattern, it creates a nice symmetry.

Reply:
FWIW, I copied the `ThinTokenStream::stream` method. But I can change it.

Reply:
I removed `stream` and used `TokenStream::Tree` throught.